### PR TITLE
feat(manager/pip-compile): Create wrapper for pip extractPackageFile

### PR DESCRIPTION
**Pull request by not7cd**
This is in preparation for #26858. 

## Changes

`pip-compile` lock files can be built from multiple sources, and not only from `requirements.in`. Original `pip_requirements.extractPackageFile()` has been wrapped with switch case statements that check if source file could be handled by `setuptools` managers or `pep621`.

This change shouldn't be breaking, as it will have no direct effect in current constraints. Namely, `.in-.txt` pairing between files. 

## Context

#26858 too big

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/pip-compile/types.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// managers supported by pip-tools Python package
+export type SupportedManagers =
```
Comment (rarkins) [inline lib/modules/manager/pip-compile/types.ts:2]: Not sure this is supported

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — DISMISSED** [review]

**Review by viceice — COMMENTED** [review]
just a small nit

File: lib/modules/manager/pip-compile/extract.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    return 'pip_setup';
+  }
+  if (filename.endsWith('setup.cfg')) {
+    return 'setup-cfg';
+  }
+  if (filename.endsWith('pyproject.toml')) {
+    return 'pep621';
+  }
+  // naive, could be improved, maybe use pip_requirements.fileMatch
+  if (filename.endsWith('.in')) {
```
Comment (viceice) [inline lib/modules/manager/pip-compile/extract.ts:20]: 🤔 it's a kind of breaking when users didn't use `.in` files already as input? can we treat unknown as warning and fallback to requirements for now?

**Review by viceice — DISMISSED** [review]

File: lib/modules/manager/pip-compile/extract.ts (hunk 2 above)
Comment (not7cd) [reply lib/modules/manager/pip-compile/extract.ts:20]: let's fallback

**Review by viceice — APPROVED** [review]

**Pull request merged**